### ISSUE TITLE
`R_LIBS_USER` not blocked for conda environments

### ISSUE THREAD

**Issue by jashapiro** [snakemake/snakemake#2163]
**Snakemake version**
7.24.0

**Describe the bug**
When using a conda environment, snakemake helpfully disables the `R_LIBS` environment variables to prevent loading conflicting packages. However, on some systems `R_LIBS_USER` is also a potential source of package conflicts, so this environment variable should perhaps be blocked as well. 

**Logs**

**Minimal example**
I am happy to provide an example, but since it involves using a conda environment, I wasn't sure the best way to go about adding one here.

I will note that adding the following line to the `.post-deploy.sh` script for the environment solves the issue on a case-by-case basis.

```
conda env config vars set R_LIBS_USER=''
```

**Additional context**

I think that adding `R_LIBS_USER` to the list in the following code chunk would solve the issue. 

https://github.com/snakemake/snakemake/blob/c7ea0590c396c67fa5d56042e21f678c20784d3b/snakemake/shell.py#L246-L252

It may also be useful to unset `R_LIBS_SITE` though my impression is that is used much less frequently.

**Comment by mfansler** [comment]
Sounds more like a feature request than a bug report.

One working alternative is to include the `conda-ecosystem-user-package-isolation` package in the Conda environment. Some discussion here: https://github.com/conda-forge/r-base-feedstock/pull/65#issuecomment-1098457402.

Another option is that YAMLs support environment variable definitions directly. See [docs](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#setting-environment-variables).

**Comment by jashapiro** [comment]
> Sounds more like a feature request than a bug report.

Yes, I went back and forth, but I thought that since environment isolation was already an advertised feature, it might slot in this way. 

> One working alternative is to include the `conda-ecosystem-user-package-isolation` package in the Conda environment. Some discussion here: [conda-forge/r-base-feedstock#65 (comment)](https://github.com/conda-forge/r-base-feedstock/pull/65#issuecomment-1098457402).
> 
> Another option is that YAMLs support environment variable definitions directly. See [docs](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#setting-environment-variables).

Thank you for pointing me to these options! I had somehow not gotten far enough down in the documentation to see the YAML version of ENV setting; I guess I stopped when I found the `conda env config vars` settings. 

I will also look into the `conda-ecosystem-user-package-isolation`, as that seems like it may be more robust and complete, especially in the context of a workflow manager. Thanks again for the suggestions!
